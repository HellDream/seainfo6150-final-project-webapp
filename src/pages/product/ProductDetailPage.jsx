import React, { useEffect, useState } from 'react';
import styles from './ProductDetailPage.module.css';
import Header from '../../components/headers/Header.jsx';
import { isEmpty } from 'lodash';
import dayjs from 'dayjs';
import relativeTime from 'dayjs/plugin/relativeTime';

const ProductPage = (props) => {
    dayjs.extend(relativeTime);

    const [product, setProduct] = useState();
    useEffect(() => {
        document.title = "Thrift Store - Product Detail";

        const fetchData = async () => {
            const res = await fetch(
                'https://us-central1-seainfo6150-final-project.cloudfunctions.net/api/getProductById',
                {
                    method: 'POST',
                    headers: {
                        'Content-Type': 'application/json',
                    },
                    body: JSON.stringify({
                        category: props.categoryTitle,
                        slug: props.productId,
                    }),
                }
            );
            const json = await res.json();
            setProduct(json);
        };

        if (isEmpty(product)) {
            if (props.product) {
                setProduct(props.product);
            } else {
                fetchData();
            }
        }
    }, [product]);

    return isEmpty(product) ? (
        <div />
    ) : (
        <>
            <Header />
            <section className={styles.section}>
                <div className={styles.container}>
                    <div className={styles.imageContainer}>
                        <img
                            className={styles.image}
                            src={product.imageUrl}
                            alt="book"
                        />
                    </div>
                    <div className={styles.titleContainer}>
                        <h3 className={styles.title}>{product.title}</h3>
                        <h4 className={styles.price}>
                            {product.price.toFixed(2)}
                        </h4>
                    </div>
                    <div className={styles.contactContainer}>
                        <address className={styles.contact}>
                            Posted by {product.contactName + ' '}
                            <a href="mailto:kate.farley@nytimes.com">
                                {product.email}
                            </a>
                        </address>
                        <time dateTime={product.postedTime}>
                            {dayjs(product.postedTime).fromNow()}{' '}
                        </time>
                    </div>
                    <div className={styles.descriptionContainer}>
                        <p>{product.description}</p>
                    </div>
                </div>
            </section>
        </>
    );
};

export default ProductPage;
