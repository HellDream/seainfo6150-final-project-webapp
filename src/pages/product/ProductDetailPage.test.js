import React from 'react';
import ProductDetailPage from './ProductDetailPage';
import { render } from '@testing-library/react';

import img from '../../images/books.jpg';
import { BrowserRouter } from 'react-router-dom';
const product = {
    slug: 1,
    imageUrl: img,
    title: 'book',
    postedTime: new Date('05 October 2011 14:48 UTC').toISOString(),
    price: 2.33,
    description: 'description',
};
describe('CategoryItem tests', () => {
    it('renders correctly', () => {
        const { container } = render(
            <BrowserRouter>
                <ProductDetailPage categoryTitle="books" product={product} />
            </BrowserRouter>
        );
        expect(container).toMatchSnapshot();
    });
});
