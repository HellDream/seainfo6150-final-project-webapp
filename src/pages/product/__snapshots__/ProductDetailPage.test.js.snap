// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryItem tests renders correctly 1`] = `
<div>
  <div>
    <div />
    <header
      class="header"
    >
      <div
        class="navbar"
      >
        <div
          class="backContainer"
        >
          <button
            class="backBtn"
          />
        </div>
        <a
          class="homeLink"
          href="/"
        >
          <h3
            class="title"
          >
            thrift store
          </h3>
        </a>
        <div
          class="links"
        >
          <a
            class="createLink"
            href="/create"
          >
            create
          </a>
        </div>
      </div>
    </header>
  </div>
  <section
    class="section"
  >
    <div
      class="container"
    >
      <div
        class="imageContainer"
      >
        <img
          alt="book"
          class="image"
          src="books.jpg"
        />
      </div>
      <div
        class="titleContainer"
      >
        <h3
          class="title"
        >
          book
        </h3>
        <h4
          class="price"
        >
          $
          2.33
        </h4>
      </div>
      <div
        class="contactContainer"
      >
        <address
          class="contact"
        >
          Posted by 
          undefined 
          <a
            href="mailto:kate.farley@nytimes.com"
          />
        </address>
        <time
          datetime="2011-10-05T14:48:00.000Z"
        >
          9 years ago
           
        </time>
      </div>
      <div
        class="descriptionContainer"
      >
        <p>
          description
        </p>
      </div>
    </div>
  </section>
  <footer
    class="container"
  >
    <a
      class="content"
      href="/contact-us"
    >
      Contact us?
    </a>
  </footer>
</div>
`;
